### ISSUE TITLE
3方纵向LR，有一方host端reader数据出错

### ISSUE THREAD

**Issue by tigflanker** [FederatedAI/FATE#5495]
使用的是KubeFATE，版本1.11.1

提交任务采用的是 "dsl_version": "2" 方式，想要跑通的实例是 fateflow/examples/lr/test_hetero_lr_job_conf.json

guest：xx.xx.xx.163、host1：xx.xx.xx.162、host2：xx.xx.xx.129

guest正常上传数据：flow data upload -c fateflow/examples/upload/upload_guest.json
host1上传数据时，将 table_name改为 "table_name": "breast_hetero_host1"
host2上传时，将 table_name改为 "table_name": "breast_hetero_host2"

然后在提交测试任务的时候，可能是我的conf写的不对，在host1（和host2）看板上显示的数据集均为：
```
HOST 10000
dataset experiment.breast_hetero_host2
```

最终在hetero_lr_0上面报错了，最后一条info是：
[INFO] [2024-02-22 09:56:14,783] [202402220954172005300] [33597:140139433416512] - [hetero_lr_gradient_and_loss.compute_loss] [line:128]: More than one host exist, loss is not available

修改过后的config、以及提交信息是这样的：
```
vi fateflow/examples/lr/test_hetero_lr_job_conf_3p.json
ggdGi

{
  "dsl_version": "2",
  "initiator": {
    "role": "guest",
    "party_id": 9999
  },
  "role": {
    "guest": [
      9999
    ],
    "host": [
      10000, 9998
    ],
    "arbiter": [
      10000
    ]
  },
  "job_parameters": {
    "common": {
      "task_parallelism": 2,
      "computing_partitions": 8,
      "task_cores": 4,
      "auto_retries": 1
    }
  },
  "component_parameters": {
    "common": {
      "intersection_0": {
        "intersect_method": "raw",
        "sync_intersect_ids": true,
        "only_output_key": false
      },
      "hetero_lr_0": {
        "penalty": "L2",
        "optimizer": "rmsprop",
        "alpha": 0.01,
        "max_iter": 3,
        "batch_size": 320,
        "learning_rate": 0.15,
        "init_param": {
          "init_method": "random_uniform"
        }
      }
    },
    "role": {
      "guest": {
        "0": {
          "reader_0": {
            "table": {
              "name": "breast_hetero_guest",
              "namespace": "experiment"
            }
          },
          "data_transform_0": {
            "with_label": true,
            "label_name": "y",
            "label_type": "int",
            "output_format": "dense"
          }
        }
      },
      "host": {
        "0": {
          "reader_0": {
            "table": {
              "name": "breast_hetero_host1",
              "namespace": "experiment"
            }
          },
          "data_transform_0": {
            "with_label": false,
            "output_format": "dense"
          },
          "evaluation_0": {
            "need_run": false
          }
        },
        "1": {
          "reader_0": {
            "table": {
              "name": "breast_hetero_host2",
              "namespace": "experiment"
            }
          },
          "data_transform_0": {
            "with_label": false,
            "output_format": "dense"
          },
          "evaluation_0": {
            "need_run": false
          }
        }
      }
    }
  }
}

flow job submit -d fateflow/examples/lr/test_hetero_lr_job_dsl.json -c fateflow/examples/lr/test_hetero_lr_job_conf_3p.json
```

**Comment by mgqa34** [comment]
多host的时候，lr不计算loss，需要使用weight_diff去做收敛判断

**Comment by tigflanker** [comment]
@mgqa34 感谢。改用secure boost至少已经成功了，我再试试LR。

另外上传数据这个dashboard的显示可能是关联错了，10000应该显示 host1.test_host1的数据集，结果是正确的
![bug](https://github.com/FederatedAI/FATE/assets/22868157/be9b2025-7dd3-4be7-a154-128aa1aa7ebf)

**Comment by hztBUAA** [comment]
> @mgqa34 感谢。改用secure boost至少已经成功了，我再试试LR。
> 
> 另外上传数据这个dashboard的显示可能是关联错了，10000应该显示 host1.test_host1的数据集，结果是正确的 ![bug](https://private-user-images.githubusercontent.com/22868157/306905246-be9b2025-7dd3-4be7-a154-128aa1aa7ebf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM1MzQ2NzAsIm5iZiI6MTcxMzUzNDM3MCwicGF0aCI6Ii8yMjg2ODE1Ny8zMDY5MDUyNDYtYmU5YjIwMjUtN2RkMy00YmU3LWExNTQtMTI4YWExYWE3ZWJmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDE5VDEzNDYxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUxMzYwZThkNzhiZThmMTlmMzYzZTgzOGMwYzBmYmI5N2YxYmMyMDNjZWI3YjQzZjdhM2Q5ZTk2OGM3MmFlMGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.GpamveXMTGFVuYkgvNHK9zoo69-OG8sR_Vq1Pzh-z4A)

你好，想问一下看板上两个host显示的数据集是一样的这个正常吗，但是我的conf配置信息显示两个host的数据集是不同的。然后运行时也是无法正常运行，在secure boot模型fit时出错了
